### DISCUSSION TITLE
Is it able to make pane grid something like `PaneGrid` in `iced-rs`?

### DISCUSSION THREAD

**Discussion by ArtBlnd** [slint-ui/slint#5221] [Q&A]
![image](https://camo.githubusercontent.com/cc593483384060029bfb9c0238ececa8282502606aad4b0a0cd07ddd2f9bebb3/68747470733a2f2f696365642e72732f6578616d706c65732f70616e655f677269642e676966)

I need widget something like this. Is it possible to make widget like this using slint?

**Comment by ghost** [comment]
Hi,
there is nothing builtin yet in Slint, but I think it should be possible to create such a feature. What you need for this is some sort of drag and drop. I have done drag and drop for my custom widgets https://codeberg.org/flovansl/co_sl/src/branch/main/widgets/coop/ui/extensions/list_view_extension.slint, https://codeberg.org/flovansl/co_sl/src/branch/main/widgets/coop/ui/widgets/standard_list_view.slint.

The rest could looks like this:

```slint
export component MyContainer {
      in property <string> title;
      in property <int> pane-row;
      in property <int> pane-column;
      // ....
}
```

```slint
export component PaneGrid {
    in property <[ContainerModel]> model;
    
    ...
    
    for model[index] in root.model : MyContainer {
          x: root.get-x(index);
          y: root.get-y(index);
          
          title: model.title;
    }
    
    function get-x(index: int) -> length {
         // calculate x for the item on the given index from the model
     }
     
      function get-y(index: int) -> length {
         // calculate y for the item on the given index from the model
     }
}
```

**Reply by ArtBlnd** [reply]
Can't we just hold container itself using property?

**Reply by ghost** [reply]
What do you mean. Can you give me a small code example?

**Reply by ArtBlnd** [reply]
Like
```
in property <component> components;
```

**Reply by ghost** [reply]
Ah I see what you mean. At the moment that is not available in the public api. There is a type called `ComponentFactory` that can be used to integrate components dynamically in Slint. It is e.g. used for the preview. But this feature is experimental.

**Reply by ArtBlnd** [reply]
I tested `ComponentFactory` by enabling experimental features. However, it looks like the layout computation on the `ComponentFactory` does not work properly.

**Reply by ArtBlnd** [reply]
How long should I wait for the feature?

**Reply by ArtBlnd** [reply]
Is any way to implement it without `ComponentFactory`? Like using `private_unstable_api`

**Reply by ghost** [reply]
@hunger can you something about the state of `ComponentFactory`?

**Reply by ogoffart** [reply]
I'm not sure how soon ComponentFactory will be ready, but i wouldn't hold my breath.

In the mean time, the best is to just have the data in the model, and do something like:

```slint
if component.type == 0 : ComponentFoo { ... }
if component.type == 1 : ComponentBar { ... }
if component.type == 2 : ComponentZib { ... }
...
```
